src/main/java/javaslang/control/Failure.java

```diff
@@ -150,10 +150,16 @@ public void orElseRun(Consumer<? super Throwable> action) {
 
     @SuppressWarnings("unchecked")
     @Override
-    public <R> Failure<R> andThen(CheckedFunction1<T, R> f) {
+    public <R> Failure<R> then(CheckedFunction1<T, R> f) {
         return (Failure<R>) this;
     }
 
+    @SuppressWarnings("unchecked")
+    @Override
+    public Failure<T> thenRun(CheckedConsumer<T> f) {
```

Comment:
Will be `andThen(CheckedConsumer<? super T> f)`. Please note the generic type bounds.